benchmarking/generate_users.py

```diff
@@ -0,0 +1,67 @@
+"""
+generates dummy users by sending multiple concurrent requests to signup API
+stores the user details like UserID, Username, EmailID, MobileNo, Password,
+Response Time and Response Status Code to a CSV file
+"""
+
+import concurrent.futures
+import random_string
+import requests
+import csv
+
+USERNAME_MIN_LENGTH = 4
+USERNAME_MAX_LENGTH = 8
+PASSWORD_LENGTH = 8
+MOBILE_NUMBER_LENGTH = 10
+EMAIL_DOMAIN = "@dummy.com"
+ITERATIONS = 1000000
+OUTPUT_FILE_NAME = "users.csv"
+CSV_COLUMN_NAMES = ["UserID", "Username", "EmailID", "Password", "MobileNo", "Response Time"]
+
+
+def signup_request():
+    username = random_string.alpha_numeric_variable_length(USERNAME_MIN_LENGTH, USERNAME_MAX_LENGTH)
+    email = username + EMAIL_DOMAIN
+    mobile_no = random_string.numeric_fixed_length(MOBILE_NUMBER_LENGTH)
+    password = random_string.alpha_numeric_fixed_length(PASSWORD_LENGTH)
+    response = requests.post('https://gcp-chat-service.an.r.appspot.com/signup',
+                             json={"Username": username,
+                                   "EmailID": email,
+                                   "MobileNo": mobile_no,
+                                   "Password": password,
+                                   }
+                             )
+    while response.status_code != 200:
```

Comment:
what if it is a non-retryable error?
In that case this will loop forever.
There should be some exit condition like max number of retries or something.

Reply:
Added maximum tries as 5.